cmd/cni-metrics-helper/metrics/metrics.go

```diff
@@ -412,13 +416,13 @@ func metricsListGrabAggregateConvert(t metricsTarget) (map[string]*dto.MetricFam
 		origFamilies, err := parser.TextToMetricFamilies(bytes.NewReader(rawOutput))
 
 		if err != nil {
-			glog.Warning("Failed to parse metrics:", err)
+			log.Warnf("Failed to parse metrics:", zap.Error(err))
```

Comment:
It's best practice to simply return the error and let the top-most caller (i.e. `main()`) determine whether to print a message to a log.

The reason for this best practice is to control what to do with errors (log them, aggregate them, forward them, etc) in a single place.

Read here for more information on this: https://dave.cheney.net/2015/11/05/lets-talk-about-logging

In particular the following:

>  If an error that cannot be handled occurs, it should bubble up main.main where the program terminates. The minor inconvenience of having to insert the FATAL prefix in front of the final log message, or writing directly to os.Stderr with fmt.Fprintf is not sufficient justification for a logging package growing a log.Fatal method.